sympy/sets/sets.py

```diff
@@ -1202,28 +1202,32 @@ def reduce(args):
 
         # ===== Pair-wise Rules =====
         # Here we depend on rules built into the constituent sets
-        args = set(args)
+        args = FiniteSet(*args)
```

Comment:
`FiniteSet` doesn't guarantee an order. A tuple is already ordered, you can create a copy of `args` and use it later where you need it.

_General Comment_: You should avoid using undocumented behavior  of any function as they might change in future. For example the current algorithm used in converting list to `FiniteSet` might maintain order but a future implementations might change this behavior breaking your code.

/cc @kshitij10496